pkg/utils/selector.go

```diff
@@ -90,8 +90,12 @@ func selectPods(ctx context.Context, c client.Client, selector v1alpha1.Selector
 				if err != nil {
 					return nil, err
 				}
-
-				pods = append(pods, pod)
+				if IsAllowedNamespaces(pod.Namespace) {
+					pods = append(pods, pod)
+				} else {
+					log.Info("filter pod by namespaces",
+						"pod", pod.Name, "namespace", pod.Namespace)
+				}
```

Comment:
I prefer we don't check namespace here since there is also a check at line 122.

Or maybe you can keep this and use pods = append(pods, filterByNamespaces(podList.Items)...)

Reply:
the logic for checking namespace here is different from that in line 122.

Reply:
What is the difference? I can only see that the namespace of pods is checked twice

Reply:
https://github.com/pingcap/chaos-mesh/blob/f22ba60db393d0643597ad6f131b068fa43a2152/pkg/utils/selector.go#L82-L106

if `len(selector.Pods) > 0` , We will ignore the other options. So I think we should do a check here as well.

Reply:
Hmm, that's true. But seems the default behavior is a little weird. If we set the podSelector, then we don't do other filterings? @cwen0